Clarify how access happens for shares without absolute URIs (#165)
cs3org/OCM-API: IETF-RFC.md

* __Discovering Server__ - a server that tries to obtain information in OCM API discovery
* __Discoverable Server__ - a server that tries to supply information in OCM API discovery
* __OCM Address__ - a string of the form `<Receiving Party's identifier>@<fqdn>` which can be used to uniquely identify a user or group "at" an OCM Server. `<Receiving Party's identifier>` is an opaque string,

> unique at the server. `<fqdn>` is the Fully Qualified Domain Name by which the server is identified. This MUST be the domain at which the OCM API of that server is hosted.

Author: Reviewing myself, I believe this is the only significant change, all the rest being a matter of better phrasing and realigning of the IETF text to the spec.

I believe we have to force a `MUST` here, otherwise my argument of un-deprecating full URIs in share payloads remains valid. Otherwise, if a share payload does not contain a usable owner, and the URI is relative, how can a receiver contact the sender, assuming that its remote address may not be valid?

Reviewer: I think this "doesn't have to be" phrase refers to the fact we are allowing the OCM API  (i.e. where `POST ./shares` and other endpoints live) as advertised in `/.well-known/ocm` to be on a different server than `/.well-known/ocm` itself, right?

Author: Ah, my interpretation was that the `/.well-known/ocm` must be on the same server as the owner's (or recipient's) OCM address that is sent by a `POST ./shares`

Reviewer: ```suggestion
unique at the server. `<fqdn>` is the Fully Qualified Domain Name by which the server is identified. This MUST be the domain at which the `/.well-known/ocm` endpoint of that server is hosted.
```

> Step 1: In case it has an OCM Address, it should first extract `<fqdn>` from it (the part after the `@` sign).

Author: ```suggestion
Step 1: In case it has an OCM Address, it should first extract `<fqdn>` from it (the part after the last `@` sign).
```

                only support `webdav`.

   * Protocol details for `webdav` MAY contain:

>      * REQUIRED uri (string)

Reviewer: why is this one moving from optional to required, was that discussed?

Author: In fact it has always been required (in spec.yaml), the IETF text was actually incorrect

                      - `use-code` requires the consumer to exchange the given `code` via a
                        signed HTTPS request. This MAY be used if the recipient provider exposes
                        the `receive-code` capability.
   * Protocol details for `webapp` MAY contain:

>      * REQUIRED uri (string)

Reviewer: is this a breaking change?

Author: (see previous comment, it's not a change)

# Resource Access
To access the Resource, the Receiving Server MAY use multiple ways, depending on the body of the Share Creation Notification. The procedure is as follows:

> 1. The receiver MUST extract the OCM Server FQDN from the `owner` field of the received share, and MUST query the [Discovery](#ocm-api-discovery) endpoint at that address: the `resourceTypes[0].protocols.webdav` value is the `<sender-ocm-path>` to be used in step 3.

Reviewer: `sender`, not `owner`, right?
```suggestion
1. The receiver MUST extract the OCM Server FQDN from the `sender` field of the received share, and MUST query the [Discovery](#ocm-api-discovery) endpoint at that address: the `resourceTypes[0].protocols.webdav` value is the `<sender-ocm-path>` to be used in step 3.
```
